### PR TITLE
Add ssh::config_entry defined type

### PR DESCRIPTION
**Pull request by jeffmccune**
This patch adds a mechanism to manage ~/.ssh/config entries using the
concat module.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ ssh_key_ensure and purge_keys.
 
 This module may be used with a simple `include ::ssh`
 
+The `ssh::config_entry` defined type may be used directly and is used to manage
+Host entries in a personal `~/.ssh/config` file.
+
```
Comment (ghoneycutt) [inline README.md]: @Phil-Friderici We could use a param `ssh::config_entries` that is a hash that is used with create_resources for ssh::config_entry resources.

File: manifests/config_entry.pp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# == Define: ssh::config_entry
+#
+# Manage an entry in ~/.ssh/config for a particular user.  Lines model the lines
+# in each Host block.
+define ssh::config_entry (
```
Comment (ghoneycutt) [inline manifests/config_entry.pp]: @Phil-Friderici we could also use spec tests. This is for our friends in NY.

File: README.md (hunk 1 above)
Comment (Phil-Friderici) [reply README.md]: sure, like in the types and other modules

File: manifests/config_entry.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/config_entry.pp]: will be my pleasure

**Pull request merged**